app/api/schema/discount_codes.py

```diff
@@ -245,7 +248,10 @@ def validate_date(self, data, original_data):
             if 'valid_till' not in data:
                 data['valid_till'] = discount_code.valid_till
 
-        if data['valid_from'] >= data['valid_till']:
+            if not data['valid_till']:
+                data['valid_till'] = discount_code.valid_expire_time
```

Comment:
Defeats the purpose as previously reviewed. Now the time is static. We don't want to store the end time if when sales ends at is selected.

Ask in a call if you are still confused